app/src/main/java/com/simply/birthdayapp/main/addEvent/presentation/AddEventViewModel.kt

```diff
@@ -88,9 +104,58 @@ class AddEventViewModel : ViewModel() {
         }
     }
 
-    fun setImageUrl(newValue: Uri?) {
+    fun setImageUri(newValue: Uri?) {
+        viewModelScope.launch {
+            _imageSource.value = if (newValue != null) {
+                ImageSource.Uri(newValue.toString())
+            } else {
+                ImageSource.Unknown
+            }
+        }
+    }
+
+    private fun createDateFromSelectedValues(selectedYear: Int, selectedMonth: Int, selectedDay: Int): Date {
+        val calendar = Calendar.getInstance().apply {
+            set(Calendar.YEAR, selectedYear)
+            set(Calendar.MONTH, selectedMonth - 1)
```

Comment:
Here -1 is used because in Calendar.MONTH months are counted from 0, not 1.

When you pass selectedMonth, it is a value between 1 and 12, but Calendar expects values ​​between 0 (January) and 11 (December). So you need to subtract 1 to bring the selectedMonth value into the format that Calendar expects.